### ISSUE TITLE
AttributeError GPCCA compute_microstates

### ISSUE THREAD

**Issue by jonas2612** [theislab/cellrank#1097]
GPCCA.compute_microstates doesn't work with pandas 2.0.1

```python
import scanpy as sc
import scvelo as scv
from anndata import concat
from cellrank.kernels import VelocityKernel
from cellrank.estimators import GPCCA

adata = scv.datasets.gastrulation_erythroid()
adata = adata[adata.obs['stage'].isin(['E8.0', 'E8.5'])]
sc.pp.log1p(adata)
sc.pp.neighbors(adata)
sc.pp.highly_variable_genes(adata, n_top_genes=2000)

adata_1 = adata[adata.obs['stage'] == 'E8.0']
adata_2 = adata[adata.obs['stage'] == 'E8.5']

sc.pp.pca(adata_1, n_comps=50)
scv.pp.moments(adata_1)
scv.tl.velocity(adata_1)

sc.pp.pca(adata_2, n_comps=50)
scv.pp.moments(adata_2)
scv.tl.velocity(adata_2)

adata.layers['velocity'] = concat([adata_1, adata_2]).layers['velocity']
adata.layers['Mu'] = concat([adata_1, adata_2]).layers['Mu']
adata.layers['Ms'] = concat([adata_1, adata_2]).layers['Ms']

vk = VelocityKernel(adata).compute_transition_matrix()
g = GPCCA(vk)
g.compute_macrostates(n_states=5, cluster_key="celltype")
```

```pytb
AttributeError                            Traceback (most recent call last)
Cell In[3], line 24
     22 vk = VelocityKernel(adata).compute_transition_matrix()
     23 g = GPCCA(vk)
---> 24 g.compute_macrostates(n_states=5, cluster_key="celltype")

File [~/Desktop/gitrepos/cellrank/cellrank/estimators/terminal_states/_gpcca.py:211](https://file+.vscode-resource.vscode-cdn.net/Users/jonasflor/Desktop/lsModel/tutorial/~/Desktop/gitrepos/cellrank/cellrank/estimators/terminal_states/_gpcca.py:211), in GPCCA.compute_macrostates(self, n_states, n_cells, cluster_key, **kwargs)
    205     logg.warning(
    206         f"Unable to compute macrostates with `n_states={n_states}` because it will "
    207         f"split complex conjugate eigenvalues. Using `n_states={n_states + 1}`"
    208     )
    209     self._gpcca = self._gpcca.optimize(m=n_states + 1)
--> 211 self._set_macrostates(
    212     memberships=self._gpcca.memberships,
    213     n_cells=n_cells,
    214     cluster_key=cluster_key,
    215     params=self._create_params(),
    216     time=start,
    217 )
    218 return self

File [~/Desktop/gitrepos/cellrank/cellrank/estimators/terminal_states/_gpcca.py:1127](https://file+.vscode-resource.vscode-cdn.net/Users/jonasflor/Desktop/lsModel/tutorial/~/Desktop/gitrepos/cellrank/cellrank/estimators/terminal_states/_gpcca.py:1127), in GPCCA._set_macrostates(self, memberships, n_cells, cluster_key, check_row_sums, time, params)
   1124 self._write_states("initial", states=None, colors=None, probs=None, memberships=None, log=False)
   1125 self._write_states("terminal", states=None, colors=None, probs=None, memberships=None, log=False)
-> 1127 assignment, colors = self._set_categorical_labels(assignment, cluster_key=cluster_key)
   1128 memberships = Lineage(memberships, names=list(assignment.cat.categories), colors=colors)
   1129 # fmt: on

File [~/Desktop/gitrepos/cellrank/cellrank/estimators/terminal_states/_term_states_estimator.py:619](https://file+.vscode-resource.vscode-cdn.net/Users/jonasflor/Desktop/lsModel/tutorial/~/Desktop/gitrepos/cellrank/cellrank/estimators/terminal_states/_term_states_estimator.py:619), in TermStatesEstimator._set_categorical_labels(self, categories, cluster_key, existing)
    612         colors_reference = _create_categorical_colors(len(series_reference.cat.categories))
    614     names, colors = _map_names_and_colors(
    615         series_reference=series_reference,
    616         series_query=series_query,
    617         colors_reference=colors_reference,
    618     )
--> 619     categories.cat.categories = names
    620 else:
    621     colors = _create_categorical_colors(len(categories.cat.categories))

File [~/mambaforge/envs/lsmodel_pandas/lib/python3.10/site-packages/pandas/core/base.py:178](https://file+.vscode-resource.vscode-cdn.net/Users/jonasflor/Desktop/lsModel/tutorial/~/mambaforge/envs/lsmodel_pandas/lib/python3.10/site-packages/pandas/core/base.py:178), in NoNewAttributesMixin.__setattr__(self, key, value)
    172 if getattr(self, "__frozen", False) and not (
    173     key == "_cache"
    174     or key in type(self).__dict__
    175     or getattr(self, key, None) is not None
    176 ):
    177     raise AttributeError(f"You cannot add any new attribute '{key}'")
--> 178 object.__setattr__(self, key, value)

File [~/mambaforge/envs/lsmodel_pandas/lib/python3.10/site-packages/pandas/core/accessor.py:99](https://file+.vscode-resource.vscode-cdn.net/Users/jonasflor/Desktop/lsModel/tutorial/~/mambaforge/envs/lsmodel_pandas/lib/python3.10/site-packages/pandas/core/accessor.py:99), in PandasDelegate._add_delegate_accessors.._create_delegator_property.._setter(self, new_values)
     98 def _setter(self, new_values):
---> 99     return self._delegate_property_set(name, new_values)

File [~/mambaforge/envs/lsmodel_pandas/lib/python3.10/site-packages/pandas/core/arrays/categorical.py:2460](https://file+.vscode-resource.vscode-cdn.net/Users/jonasflor/Desktop/lsModel/tutorial/~/mambaforge/envs/lsmodel_pandas/lib/python3.10/site-packages/pandas/core/arrays/categorical.py:2460), in CategoricalAccessor._delegate_property_set(self, name, new_values)
   2459 def _delegate_property_set(self, name, new_values):
-> 2460     return setattr(self._parent, name, new_values)

AttributeError: can't set attribute 'categories'
```

#### Versions:

> cellrank==1.5.1+ge54987b8 scanpy==1.9.3 anndata==0.8.0 numpy==1.24.3 numba==0.57.0 scipy==1.10.1 pandas==2.0.1 pygpcca==1.0.4 scikit-learn==1.2.2 statsmodels==0.14.0 python-igraph==0.10.4 scvelo==0.2.5 pygam==0.8.0 matplotlib==3.7.1 seaborn==0.12.2

**Comment by Marius1311** [comment]
Can you try installing pyGPCCA from main? I think that's a problem with pandas 2

**Comment by jonas2612** [comment]
I just tried it and I got the same error

**Comment by jonas2612** [comment]
A fix could be replacing `categories.cat.categories = names` with `categories.cat.rename_categories(names)` in `cellrank/estimators/terminal_states/_term_states_estimator.py`, line 619. That just worked for me

**Comment by michalk8** [comment]
This has been fixed in #1079, please install `CellRank` from `main`.
